Backend/ServerManagement.Azure/AzureServerManager.cs

```diff
@@ -0,0 +1,359 @@
+            var gameInfo = ParseGameTimeResponse(response);
+            
+            return gameInfo;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Failed to get game server info");
+            throw new GameServerUnreachableException("Failed to get game server info", ex);
+        }
+    }
+
+    public async Task<IReadOnlyList<PlayerInfo>> GetPlayersAsync()
+    {
+        try
+        {
+            var command = "listplayers";
+            var response = await SendTelnetCommandAsync(command);
+            
+            // Parse the response to extract player information
+            var players = ParsePlayersResponse(response);
+            
+            return players;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Failed to get players list");
+            throw new GameServerUnreachableException("Failed to get players list", ex);
+        }
+    }
+
+    private async Task<VirtualMachineResource> GetVirtualMachineAsync()
+    {
+        var subscription = await _armClient.GetDefaultSubscriptionAsync();
+        var resourceGroup = subscription.GetResourceGroup(_vmConfig.ResourceGroupName);
+        var vm = resourceGroup.Value.GetVirtualMachine(_vmConfig.VmName);
+        return vm;
+    }
+
+    private static VmState ExtractVmStateFromInstanceView(VirtualMachineInstanceView? instanceView)
+    {
+        if (instanceView?.Statuses == null)
+            return VmState.Stopped;
+
+        var powerState = instanceView.Statuses
+            .FirstOrDefault(s => s.Code?.StartsWith("PowerState/") == true)
+            ?.Code?
+            .Replace("PowerState/", "");
+
+        return powerState?.ToLower() switch
+        {
+            "running" => VmState.Running,
+            "starting" => VmState.Starting,
+            "stopped" => VmState.Stopped,
+            "stopping" => VmState.Stopping,
+            "deallocated" => VmState.Deallocated,
+            "deallocating" => VmState.Deallocating,
+            _ => VmState.Stopped
+        };
+    }
+
+    private async Task<bool> IsGamePortOpenAsync()
+    {
+        try
+        {
+            using var tcpClient = new TcpClient();
+            var connectTask = tcpClient.ConnectAsync(_gameConfig.Host, _gameConfig.Port);
+            var timeoutTask = Task.Delay(TimeSpan.FromSeconds(5));
+            
+            var completedTask = await Task.WhenAny(connectTask, timeoutTask);
+            
+            if (completedTask == connectTask && tcpClient.Connected)
+            {
+                return true;
+            }
+            
+            return false;
+        }
+        catch
+        {
+            return false;
+        }
+    }
+
+    private async Task<string> SendTelnetCommandAsync(string command)
+    {
+        try
+        {
+            using var tcpClient = new TcpClient();
+            await tcpClient.ConnectAsync(_gameConfig.Host, _gameConfig.TelnetPort);
```

Comment:
The 7d2d server requires a telnet password. It's not anonymous login. Will this work with that?